specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,247 @@
+            .Get());
+}
+void AppWindow::OnCreateEnvironmentCompleted(
+    HRESULT result, ICoreWebView2Environment* environment)
+{
+    if (environment)
+    {
+        wil::unique_cotaskmem_string version;
+        CHECK_FAILURE(environment->get_BrowserVersionString(&version));
+        CheckChannel(version.get());
+    }
+}
+```
+# API Details
+```
+typedef enum COREWEBVIEW2_RUNTIME_CHANNEL {
+  /// The stable WebView2 Runtime that is released every 4 weeks.
+  COREWEBVIEW2_RUNTIME_CHANNEL_STABLE,
+  /// The Beta Runtime channel that is released every 4 weeks, a week before the
+  /// stable release.
+  COREWEBVIEW2_RUNTIME_CHANNEL_BETA,
+  /// The Dev Runtime channel that is released weekly.
+  COREWEBVIEW2_RUNTIME_CHANNEL_DEV,
+  /// The Canary Runtime channel that is released daily.
+  COREWEBVIEW2_RUNTIME_CHANNEL_CANARY,
+} COREWEBVIEW2_RUNTIME_CHANNEL;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_RUNTIME_CHANNEL)")
+
+/// Additional options used to create the WebView2 Environment that support
+/// specifying the `RuntimeChannelPreferenceSet`.
+[uuid(47ac856e-a726-4e04-b36b-f58dafb39e38), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+
+  /// Gets the `RuntimeChannelPreferenceSet`.
+  HRESULT GetRuntimeChannelPreferenceSet(
+      [out] COREWEBVIEW2_RUNTIME_CHANNEL* channels);
+
+  /// Sets the `RuntimeChannelPreferenceSet`, which is a mask of one or more
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL`s. OR operation(s) can be applied to multiple
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL`s to create a mask. By default, the WebView2 loader
+  /// searches for channels from most to least stable, using the first channel
+  /// found on the device. When a `RuntimeChannelPreferenceSet` is provided, the
+  /// WebView2 loader will only search for the channels specified in the set.
+  /// Set `ReverseChannelSearchOrder` to `TRUE` to reverse the search order so
+  /// that the loader searches for least stable build first. See
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL` for descriptions of each channel.
+  /// `CreateCoreWebView2EnvironmentWithOptions` fails with
+  /// `HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)` if the loader is unable to find
+  /// any channel from the `RuntimeChannelPreferenceSet` installed on the device.
+  /// Use `BrowserVersionString` on `ICoreWebView2Environment` to verify which channel
+  /// is used when this option is set.
+  ///
+  /// Examples:
+  /// |   RuntimeChannelPreferenceSet   |   Default Channel Search Order   |   Reverse Channel Search Order   |
+  /// | --- | --- | --- |
+  /// |COREWEBVIEW2_RUNTIME_CHANNEL_BETA \| COREWEBVIEW2_RUNTIME_CHANNEL_STABLE| WebView2 Runtime -> Beta | Beta -> WebView2 Runtime|
+  /// |COREWEBVIEW2_RUNTIME_CHANNEL_CANARY \| COREWEBVIEW2_RUNTIME_CHANNEL_DEV \| COREWEBVIEW2_RUNTIME_CHANNEL_BETA \| COREWEBVIEW2_RUNTIME_CHANNEL_STABLE| WebView2 Runtime -> Beta -> Dev -> Canary | Canary -> Dev -> Beta -> WebView2 Runtime |
```

Comment:
Some of the | are escaped with \ and some aren't. I don't understand why some are and some aren't? Is that intentional?

Reply:
Yes the escaped ones indicate the OR operator and the others are the column separators